docs/multi-target-framework.md

```diff
@@ -0,0 +1,150 @@
+# Multi-targeting
+
+## Developer needs / scenarios
+
+* Compile against an earlier version of our bindings.
+    * Example: a customer could be producing a NuGet targeting .NET 7, and
+      want to support the initial release of .NET 7. Said customer must then
+      be able to build against the bindings we shipped at the time.
+    * Ref: https://github.com/dotnet/sdk/issues/30103
+* Consume preview packages for a preview version of Xcode.
+    * Ref: https://github.com/xamarin/xamarin-macios/issues/18343
+
+## Developer experience
+
+### TargetFramework=net7.0-ios16.4
+
+Builds with the bindings we released for iOS 16.4 (Xcode 14.3).
+
+### TargetFramework=net7.0-ios
+
+Builds with the default bindings. This can change in any release (this is
+contrary to what other platforms do - specifically because older OS bindings
+might not work with newer Xcodes, and Apple auto-updates people's Xcodes, so
+it's rather frequent that customers use newer OS bindings. Having a lot of
+people specify the OS version in the target framework is undesirable).
+
+### TargetFramework=net8.0-ios17.0
+
+Builds with the bindings we've released for iOS 17.0 (on .NET 8).
+
+This might be preview bindings, in which case customers must also set the
+following property in their project file to make their intention clear:
+
+```xml
+<PropertyGroup>
+    <EnablePreviewFeatures>true</EnablePreviewFeatures>
+</PropertyGroup>
+```
+
+This mirrors how it's done for the Android SDK.
+
+### TargetFrameworks=net7.0-ios16.0;net8.0-ios-17.0
+
+This would only be valid for a library project.
+
+### Implementation details
+
+We'll release a single workload (per platform), with bindings for every OS
+version we currently support or plan to support. This means our stable
+releases may point to preview packages (but these preview packages have to be
+opted in by doing two things: appending the OS version to their target
+framework + setting EnablePreviewFeatures=true)
+
+We'll rename our packages. Currently we ship these packages:
+
+* Microsoft.iOS.Sdk
+* Microsoft.iOS.Ref
+* Microsoft.iOS.Runtime.ios-arm64
+* Microsoft.iOS.Runtime.iossimulator-arm64
+* Microsoft.iOS.Runtime.iossimulator-x64
+
+and now we'll add the target framework supported by these packages:
+
+* Microsoft.iOS.Sdk.net7.0_16.4
+* Microsoft.iOS.Ref.net7.0_16.4
+* Microsoft.iOS.Runtime.ios-arm64.net7.0_16.4
+* Microsoft.iOS.Runtime.iossimulator-arm64.net7.0_16.4
+* Microsoft.iOS.Runtime.iossimulator-x64.net7.0_16.4
+
+Undecided what to do about:
+
+* The template package (Microsoft.iOS.Templates)
+* The manifest package (Microsoft.NET.Sdk.iOS.Manifest-7.0.100)
```

Comment:
For these, if we think that .NET 7 should use the new Xcode / bindings by default. I would consider putting `net7.0-ios17.0` in the project template, but not change the default `$(TargetPlatformVersion)` as that might break existing projects.

Reply:
The problem is that:

* Apple auto-updates Xcode.
* Newer Xcode might not work with older versions of our SDK.
* Putting the OS version for the TargetFramework in the template might mean that `dotnet new ios` will create a project that won't compile (you'll have to modify the TargetFramework to a newer OS version, or remove it completely).

That's not a good first-use experience.

Keeping backwards compatibilty has a different consequence:

* Apple auto-updates Xcode.
* A developer's project doesn't compile anymore: they update their workloads. But it still doesn't compile... (since it's using the old SDK). They'll have to add the OS version in their TargetFramework to fix it. Note that this affects most developers (if not every single one of them).